### PR TITLE
merge cog-trt-llm into this repo

### PR DESCRIPTION
**Pull request by yorickvP**
- subtree merge cog-trt-llm
- ignore cog-trt-llm subdir in runners
- use cog-trt-llm as root dir in builder

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -38,10 +39,7 @@
         };
         # override cog.yaml:
         cog.concurrency = lib.mkForce 1;
-        # copy cog-trt-llm source into /src
-        cognix.postCopyCommands = ''
-          cp ${config.deps.cog-trt-llm}/{*.py,cog-trt-llm-config.yaml} $out/src/
-        '';
+        cognix.rootPath = lib.mkForce "${./cog-trt-llm}";
```
[the remark this answers is not in the document]
Comment (yorickvP) [reply flake.nix:42]: `/src` will be different, just containing the data from `cog-trt-llm` instead of a weird merge. But that should be it.

**Comment by joehoover** [comment]
This seems good to me. Presumably you've tested the builds?

File: flake.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (technillogue) [reply flake.nix:41]: I feel like I'd prefer to keep the weird merge as it is, I think

[the remark this answers is not in the document]
Comment (joehoover) [reply flake.nix:41]: @technillogue, could you tldr the costs/benefits as you see them? I don't think I understand your preference, but I'd like to :)

[the remark this answers is not in the document]
Comment (technillogue) [reply flake.nix:41]: I was just thinking of keeping the output images the way things are now, but I don't actually care that much

File: cog-trt-llm/.gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "tensorrtllm_backend"]
+	path = tensorrtllm_backend
+	url = https://github.com/triton-inference-server/tensorrtllm_backend.git
```
Comment (technillogue) [inline cog-trt-llm/.gitmodules:3]: we should probably drop this

File: cog-trt-llm/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+*.bin
+*.safetensors
+*.cog
+*.hypothesis
+*.pytest_cache
+__pycache__/
+*.pyc
+models/*
```
Comment (technillogue) [inline cog-trt-llm/.gitignore:8]: this doesn't do anything

**Review by technillogue — APPROVED** [review]

File: cog-trt-llm/.gitmodules (hunk 2 above)
Comment (yorickvP) [reply cog-trt-llm/.gitmodules:3]: haven't dropped it because that would break the 'legacy' dockerfile cog-trt-llm build

**Pull request merged**